src/hooks/useFetch.js

```diff
@@ -1,36 +1,10 @@
-import axios from 'axios';
-import {useEffect, useState} from 'react';
-
-export default function useFetch(url) {
-	const [state, setState] = useState({
-		data: null,
-		loading: false,
-		error: null,
-	});
-
-	useEffect(() => {
-		setState(({data}) => ({
-			data,
-			error: null,
-			loading: true,
-		}));
-		axios
-			.get(url)
-			.then(({data}) => {
-				setState({
-					error: null,
-					data,
-					loading: false,
-				});
-			})
-			.catch(error => {
-				setState(({data}) => ({
-					data,
-					error,
-					loading: false,
-				}));
-			});
-	}, [url]);
-
-	return state;
+export default function useFetch() {
+	return async (url, options) => {
+		try {
+			const result = await fetch(url, options);
+			return await result.json();
+		} catch (error) {
+			console.error(error);
+		}
+	};
```

Comment:
You can thing about to extend your error handling by  using a state. 🧐
This way you can the user a feedback, if something went wrong.  
 
Here is an example. You could implement it in your project in this way or something similar. 
Please keep attention, I didn't test the code below.
 
```
const [error, setError] = useState(null);

catch(error => {
        setError(new Error('Oooops, something went wrong.'));
        setIsLoading(false);
      });
```